### ISSUE TITLE
training data divided into two parts 

### ISSUE THREAD

**Issue by wl082013** [xinntao/BasicSR#217]
Hi, xintao, nice to see your good code.

I am using your code for some HR image segmentation problem.
I need to split the DIV2K dataset into two parts: 500 images with HR pairs, and 400 with only LR pairs. I am modifying your LQGT dataloader, however, there are some codes I cannot understand well.  in the def __get__item__ function,  I need to get 500 (LR+GT) and 400 images (LR),. 
Here are some questions for understanding your code:
(1)I saw you said you did down-sampling on the fly, what is the intention for this? 
(2) we have loaded LR image, however, why do we need to downsample img_GT to make img_LQ again? 
    # using matlab imresize
                img_LQ = util.imresize_np(img_GT, 1 / scale, True)
                if img_LQ.ndim == 2:
                    img_LQ = np.expand_dims(img_LQ, axis=2)
(3) augmentation -flip and rotate are applied to both images,  however, I am wondering dowmsampling of the HR images to small size is not an augmentatiom method?
(4) I did not see the instruction for directly using images not lmdb as inputs, so if I use images directly, how should I do it ? 
(5) For __get__item__ function, is it possible to get both 500 and 400 image through same index? 

Really appreciate for your help. Thanks a lot.

**Comment by xinntao** [comment]
1. if we only have GT images, and want to generate LR when training, it is called on-the-fly. If you have LR images before training, just ignore it.
2. These codes are in the on-the-fly scope or the `if H < GT_size or W < GT_size:`. If you have proper LR images ahead, just ignore it
3. downsampling the GT to different scales, like 0.9, 0.7. 0.6... they are also augmentation methods But you need to generate LR images under that scale.
4. just set `dataroot_LQ` and `dataroot_GT` to the image folders
5. You mean you want to get all the images in one time? Why do you need this action?

**Comment by wl082013** [comment]
Thanks xintao for your reply.
I want to read 500 images with GT pairs, and 400 images with only LR images, is it okay to modify the LQGT dataloader?  Especially in the __get_item function, the arugment index is actually interatively reading 500 with LR and HR, however, if we add 400 LR only, the index does not match, am I true? Thanks for your help.
